### DISCUSSION TITLE
Plex Provider can't determine the type of release (album/single/ep) resulting duplicate

### DISCUSSION THREAD

**Discussion by rairulyle** [music-assistant/support#3146] [Q&A] [answered]
Hi All,

It seems that the Plex provider can't determine the release type of an album. See below:

MA:
<img width="1494" alt="image" src="https://github.com/user-attachments/assets/00797a9f-e5c7-4500-b847-490ce99da14b">
<img width="1483" alt="image" src="https://github.com/user-attachments/assets/804015aa-2c2c-4cf2-af1b-45a403e09d66">

Plex:
<img width="1412" alt="image" src="https://github.com/user-attachments/assets/224226a3-8ca9-4a9f-b9b6-827a001294d2">
<img width="1401" alt="image" src="https://github.com/user-attachments/assets/6fbf19dd-2d56-480b-adcd-00bec61a2cec">

Let me know if there are anything I need to provide.

**Comment by OzGav** [comment]
I think you are talking about the same thing as this person https://github.com/music-assistant/hass-music-assistant/issues/2830

**Reply by rairulyle** [reply]
@OzGav I saw the linked issue. But I think this one is different? I am not searching and just going through the artist page. Some artists I have don’t have this issue and it gets merged automatically. See attached screenshot.

 I really think it’s the Plex Provider as you can see the Plex albums are marked as Unknown. How did i know? They look different in terms of sub text and when I filter I can see my library as Unknown type.
 
Plex: Artist • Year
<img width="1274" alt="image" src="https://github.com/user-attachments/assets/946161e3-ec1b-4867-9cf8-82e294de9ef7">

Spotify/Youtube: Album Type • Artist • Year
<img width="1279" alt="image" src="https://github.com/user-attachments/assets/04aef5da-940e-4a8e-a1ad-4c05bcdad079">


Looking through it, I also think that the Youtube Music Library have this issue that's why they don't get merged.
<img width="1258" alt="image" src="https://github.com/user-attachments/assets/8f36071d-72e7-4031-b3a8-29eab46757ff">
Resulting: Album Type • • Year


EXPECTED:  They should be merged and prioritized Library and list other provider as Other Version. Here's one artist that have no issues:
<img width="1257" alt="image" src="https://github.com/user-attachments/assets/32a98854-19b9-494c-baeb-819f618f4d62">
<img width="1274" alt="image" src="https://github.com/user-attachments/assets/05446cd9-47af-4e95-97b9-f2b98cf46c75">

**Reply by rairulyle** [reply]
Also my search is okay. I have 3 music providers. I don't see any duplicates when I search the artist so it's different.  Besides the issue, I am really thankful for making this possible to us! Let me know if you need more info about my issue above:

<img width="424" alt="image" src="https://github.com/user-attachments/assets/19eee4ac-e576-46b9-a767-8914c4f72c32">

**Reply by OzGav** [reply]
I still don’t see the problem. If you don’t see a release type that is a tagging issue. Are all your tracks tagged with Picard?

**Reply by rairulyle** [reply]
@OzGav I let Lidarr handle the metadata ([via MusicBrainz Picard script](https://github.com/RandomNinjaAtk/arr-scripts/blob/main/lidarr/readme.md)) and if Lidarr missed it Plex also handle it. As you can see, Plex does sort it well without any issue, see first image or bellow attached. If I really have to Picard all of my music metadata then what's the point of having Plex as provider then? If I did that, is it just the same of using Remote/Local Share? When using providers, I am expecting that the providers does all the stuff, so if I see it correctly on my Plex, then it should the same with MA.

![image](https://github.com/user-attachments/assets/a59d0c89-c1a9-4c7a-ad88-8a6ed6d315c3)

**Reply by OzGav** [reply]
The question about Plex is a good one. Unless there is something specific about your Plex installation that you need that MA does not provide then indeed we recommend you do not add a middle man to the process and instead use the filesystem provider.

We don't know what Plex does, maybe it queries the web for the info, maybe it makes an educated guess. All we can say is MA uses the release type tag
![image](https://github.com/user-attachments/assets/3c06f451-613c-4906-9631-3f675634c654)

Just open one of the albums in Picard and have a look at what value that tag has (or if it even exists). You can use MP3Tag as well. You just have to inspect the EXTENDED TAGS and look for MUSICBRAINZ_ALBUMTYPE

**Reply by rairulyle** [reply]
@OzGav Thanks for providing the info of which MA takes a look at. I run through with Picard but it seems like the tags have no issues (since I run scripts to auto tag them). I saw the release types of most of this artist is are tagged as ep but you can see at MA that it has no results of it and it is labelled as unknown instead. 


![image](https://github.com/user-attachments/assets/3e21fcee-9974-4aae-878a-0379bd635b32)

Let me try to use filesystem provider then instead. But this is still something worth checking as I did selected Plex expecting that what I see in Plex is what I can also see on MA.

**Reply by OzGav** [reply]
Can you confirm that it is only albums with release type EP that are a problem and that all other albums work ok?

**Reply by rairulyle** [reply]
@OzGav in this artist it is the Single and EP that doesn't have result and tagged as unknown type. In other artists, it's mixed, some of the types are tagged correctly but some are tagged as unknown as well.

I am now switching to local system and see if I'll be getting the same results.

**Reply by rairulyle** [reply]
@OzGav so it turns out that it is Plex doing that weird issue that we have. You can see that the same artist above is now being tagged and merged with the other provider. So yeah it's an issue with Plex Provider.

![image](https://github.com/user-attachments/assets/33aa28b4-4a1d-43b8-a6b9-182868f99cb9)

**Reply by OzGav** [reply]
We are doing minimal work on the Plex provider due to the lack of dev support. If the local filesystem provider is working well you are advised to use that. This issue will remain open for the time being.

edit: Actually this isn't an issue yet. If you are happy we can let this slide for now.

**Reply by rairulyle** [reply, marked as answer]
Yes. Understandable. Thanks for entertaining my post @OzGav ! Cheers!
